### DISCUSSION TITLE
How to download thumbnail pictures

### DISCUSSION THREAD

**Discussion by charlieahill** [omansak/libvideo#227] [General]
Hi, is there a way to download thumbnail images from YouTube with this package? Thanks.

**Comment by marnagy** [comment]
Hello,

I don't know if there is a way, but since in Python's PyTube module there is a way to get thumbnail, I created a simple API on cloud. The backend is stored in repository [https://github.com/marnagy/yt_backend](url), if you are interested.

Here is a way to get thumbnail as a byte array:
`async Task<byte[]> GetThumbnail(string youtubeUri)`
`{`
`var url = $"https://ytbes.herokuapp.com/thumbnail?url={youtubeUri}";`
`var pictureData = await client.GetByteArrayAsync(url);`
`return pictureData;`
`}`

where `client` is an instance of HttpClient.

**Reply by charlieahill** [reply]
I coupled it to a ByteArray to bitmap converter found here: https://qawithexperts.com/questions/29/how-to-convert-byte-array-into-image , and it works perfectly!!

Thank you very much!
